plugins/modules/purefa_volume.py

```diff
@@ -958,39 +921,47 @@ def copy_from_volume(module, array):
     """Create Volume Clone"""
     changed = False
     tgt = get_target(module, array)
-    api_version = array._list_available_rest_versions()
-    arrayv6 = get_array(module)
+    api_version = array.get_rest_version()
     if tgt is None:
         changed = True
         if not module.check_mode:
-            if DEFAULT_API_VERSION in api_version:
+            if LooseVersion(DEFAULT_API_VERSION) <= LooseVersion(api_version):
                 if module.params["add_to_pgs"]:
                     add_to_pgs = []
                     for add_pg in range(0, len(module.params["add_to_pgs"])):
                         add_to_pgs.append(
-                            flasharray.FixedReference(
-                                name=module.params["add_to_pgs"][add_pg]
-                            )
+                            FixedReference(name=module.params["add_to_pgs"][add_pg])
+                        )
+                    if LooseVersion(CONTEXT_API_VERSION) <= LooseVersion(api_version):
+                        res = array.post_volumes(
+                            with_default_protection=module.params[
+                                "with_default_protection"
+                            ],
+                            add_to_protection_groups=add_to_pgs,
+                            context_names=[module.params["context"]],
+                            names=[module.params["target"]],
+                            volume=VolumePost(
+                                source=Reference(name=module.params["name"])
+                            ),
+                        )
+                    else:
+                        res = array.post_volumes(
+                            with_default_protection=module.params[
+                                "with_default_protection"
+                            ],
+                            add_to_protection_groups=add_to_pgs,
+                            names=[module.params["target"]],
+                            volume=VolumePost(
+                                source=Reference(name=module.params["name"])
+                            ),
                         )
-                    res = arrayv6.post_volumes(
-                        with_default_protection=module.params[
-                            "with_default_protection"
-                        ],
-                        add_to_protection_groups=add_to_pgs,
-                        names=[module.params["target"]],
-                        volume=flasharray.VolumePost(
-                            source=flasharray.Reference(name=module.params["name"])
-                        ),
-                    )
                 else:
-                    res = arrayv6.post_volumes(
+                    res = array.post_volumes(
```

Comment:
Still we need IF block here - if LooseVersion(CONTEXT_API_VERSION) <= LooseVersion(api_version)
Outer loop is DEFAULT_API_VERSION, this else loop is nested one, without this cloning for fleet array with context is not working.